relay-server/src/services/buffer/mod.rs

```diff
@@ -216,6 +217,13 @@ impl EnvelopeBufferService {
         }
     }
 
+    /// Tries to evict elements from the [`PolymorphicEnvelopeBuffer`] if no capacity
+    async fn try_evict(&self, buffer: &mut PolymorphicEnvelopeBuffer) {
```

Comment:
Why does this function take `&self`?

Reply:
It does not mutate its state. Maybe I misunderstood your question.

Reply:
Ah yeah, this is a good point, I missed that, thanks for the nit!